### PR TITLE
YSHOP2-130 : COD theme > Product page > UI enhancements

### PR DESCRIPTION
**Pull request by bj-anas**
## JIRA Ticket

[YSHOP2-130](https://youcanshop.atlassian.net/browse/YSHOP2-130)

## QA Steps

-  [ ] `pnpm i`
-  [ ] `pnpm dev`

## Note

Leave empty when you have nothing to say

[YSHOP2-130]: https://youcanshop.atlassian.net/browse/YSHOP2-130?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by adildev101 — CHANGES_REQUESTED** [review]
Missing the multi column block
Screen Shot 2023-03-08 at 09 31 16

File: assets/reviews.js [hunk 1]
```diff
@@ -9,32 +9,36 @@
...
+          const reviewItem = document.createElement('li');
+
+          reviewItem.classList.add('review-item');
+          createElement(reviewItem, review);
+          reviewsWrapper.appendChild(reviewItem);
+        }
+
+        showMoreButton.style.display = "none";
+      });
+    }
```
Comment (adildev101) [inline assets/reviews.js:41]: the reviews pagination show more button should'n be just display: none.
you should add it using the sdk , we already have the pagination on the reviews.
Screen Shot 2023-03-08 at 09 35 47

Comment (bj-anas) [reply assets/reviews.js:41]: > Missing the multi column block Screen Shot 2023-03-08 at 09 31 16

it isn't included in the ticket and also we don't know how the implementation of it should be like?

Comment (bj-anas) [reply assets/reviews.js:41]: > the reviews pagination show more button should'n be just display: none. you should add it using the sdk , we already have the pagination on the reviews. Screen Shot 2023-03-08 at 09 35 47

it just a button it's one element it won't affect the performance of the web page

Comment (adildev101) [reply assets/reviews.js:41]: you are skipped the section of multi column we already have it on the home page just include it on the product page also
Screen Shot 2023-03-08 at 09 43 39

Comment (adildev101) [reply assets/reviews.js:41]: what is the purpose of adding a show more button if we just add display none display block ?

Comment (bj-anas) [reply assets/reviews.js:41]: display none and block it is a behaviour effect on the button only and for the reviews it's added from the backend

Comment (jamyoucn00) [reply assets/reviews.js:41]: For the reviews, I would say that the proper way is when the user clicks on load more button to load more views, it should call the API using the SDK

You should specify to the SDK how many reviews to return each call, check this (check this)[https://github.com/youcan-shop/youcan-js#pagination]

For the multi column block, not sure tbh, you have two choices
User will add an image in the product description in the seller area which he prefers 
Create a block that can help him doing that, like you did with the countdown

File: assets/reviews.js [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
+    const reviewItem = document.createElement('li');
+
+    reviewItem.classList.add('review-item');
+    createElement(reviewItem, review);
+    return reviewsWrapper.appendChild(reviewItem);
+  }
 
   try {
-    const reviews = await youcanjs.product.fetchReviews(productId).data();
+    const res = youcanjs.product.fetchReviews(productId, { limit: 3 });
```
Comment (ybenadda) [inline assets/reviews.js:22]: you forgot `async` here

Comment (bj-anas) [reply assets/reviews.js:22]: we have already a global async scope

Comment (ybenadda) [reply assets/reviews.js:22]: I mean isn't `youcajs.product.fetchReviews` an asynchrounous task?
why you remove `await` here?
const res = await youcanjs.product.fetchReviews(productId, { limit: 3 });

File: assets/reviews.js [hunk 3]
```diff
@@ -1,40 +1,44 @@
...
-          >
-          </div>
-        </div>
-        <div class='content'>
-          ${review.content === null ? '' : review.content}
-        </div>
-      `;
-      reviewsWrapper.appendChild(reviewItem);
-    });
+    reviews = await res.data();
```
Comment (bj-anas) [inline assets/reviews.js:24]: await is here

**Review by ybenadda — APPROVED** [review]

**Review by adildev101 — APPROVED** [review]

**Pull request merged**